.github/workflows/grapl-lint.yml

```diff
@@ -0,0 +1,48 @@
+name: Grapl Lint
+
+on:
+  check_run:
+    types: [created, rerequested]
+
+jobs:
+  rust-lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Check rust formatting
+        run: |
+          cd src/rust
+          cargo fmt -- --check
+
+  python-lint:
```

Comment:
I think grapl_analyzerlib does this as a precommit hook, we'd probably want tests and linters running for an PR, not just deployment

Reply:
I wrote this to trigger on the `check_run` event, which I *think* will fire every time a github check is executed. So if we configure a check, set branch protection on that check, then every push (e.g. to a PR branch) will trigger a `check_run` and run the lints. At least that's how I was hoping it works.